src/clj/muuntaja/parse.clj

```diff
@@ -1,21 +1,22 @@
 (ns muuntaja.parse
   (:require [clojure.string :as str])
-  (:import (com.fasterxml.jackson.databind.util LRUMap)))
+  (:import (java.util.concurrent ConcurrentHashMap)))
 
 ;;
-;; Utils
+;; Cache
 ;;
 
-(defn cache [n]
-  (LRUMap. n n))
-
-(defn fast-memoize [^LRUMap cache f]
-  (fn [& args]
-    (or (.get cache args)
-        (let [ret (apply f args)]
-          (when ret
-            (.put cache args ret))
-          ret))))
+(defn fast-memoize [size f]
```

Comment:
I wonder if this is any faster than just using `clojure.core.memoize/lru`. The concurrency code looks here OK – `((fast-memoize _ f) ...)` will always return `(f ...)` even though it may do some wasted work.